src/parsers/espresso/parser.js

```diff
@@ -0,0 +1,255 @@
+import { ATOMIC_COORD_UNITS, coefficients } from "@exabyte-io/code.js/dist/constants";
+import { mix } from "mixwith";
+
+import { primitiveCell } from "../../cell/primitive_cell";
+import { Lattice } from "../../lattice/lattice";
+import math from "../../math";
+import { MaterialParser } from "../structure";
+import { FortranParserMixin } from "../utils/fortran";
+import { IBRAV_TO_LATTICE_TYPE_MAP, regex } from "./settings";
+
+export class ESPRESSOMaterialParser extends mix(MaterialParser).with(FortranParserMixin) {
+    parse(content) {
+        this.data = this.parseNamelists(content);
+        return this.parseMaterial();
+    }
+
+    getCell() {
+        return this.getCellConfig(this.data.cards);
+    }
+
+    getElements() {
+        const { elements } = this.getAtomicPositions(this.data.cards);
+        return elements;
+    }
+
+    getCoordinates() {
+        const { coordinates } = this.getAtomicPositions(this.data.cards);
```

Comment:
This doesn't make sense - actually split lines 203-210 here, and similarly for elements, etc.

Reply:
It did make sense when the idea of intermediate format was still around